### DISCUSSION TITLE
Extensibility for inline actions in the variations table

### DISCUSSION THREAD

**Discussion by nathanss** [woocommerce/woocommerce#47777] [WooCommerce New Product Editor]
I started working on [Enable inline actions in the variations table](https://github.com/woocommerce/woocommerce/issues/40351) some days ago. There are some concerns regarding extensibility, especially for the Prices and Stock part of the epic, since the UI is closely related to the content of blocks in other parts of the application. The "Stock" and "Price" quick actions must be extensible.

In [Create VariationQuickUpdateGroup and move list price related menu items and groups to block](https://github.com/woocommerce/woocommerce/pull/47292), I explored moving the content of List Price in the variations table to a Slot Fill. That way, if the list price block gets removed, the "list price" related buttons are also removed, and that logic could be expanded to the other parts. Using this alongside SlotFills would allow us to make those quick actions extensible. That is one option, with the downside of needing to have the "parent" block rendered for the related content to appear. The application currently renders all the blocks regardless of visibility, but if we ever remove this behavior, we would have to change this. [Matt has an alternate idea to not have to render the block](https://github.com/woocommerce/woocommerce/pull/47292#discussion_r1608854039).

Another option would be exploring ways of having these fields in the template somehow. (@mdperez86 can you please share your idea in the comments?)

I'm in favor of continuing the SlotFills approach, either depending on the block render itself or, if we think that will become a problem in the future, using @mattsherman's idea of having a component with a hook that checks the layout template to see if it should render itself.

**Comment by mdperez86** [comment]
I was thinking about having embed templates so a particular block can be a modal or a popover or something that can be filled with new template blocks or with an existing part of the template. So that the popover can be opened from another block.
Following this we are not going to compromise or rework on the extensibility point since we will be covered by the templating system.

**Reply by mdperez86** [reply]
I was playing around this in this [PR](https://github.com/woocommerce/woocommerce/pull/40792)

**Comment by mattsherman** [comment]
I think that the UI for the inline actions for the variations table should be composed with the same blocks and extensibility that the main form has. Much as @mdperez86 mentions.

**Comment by mattsherman** [comment]
The quick actions are different, as they are not currently built up from blocks. So, a slot-fill approach seems to make sense.

But, I wonder what a solution where the quick actions were composed of blocks could look like and if it makes any sense. It's something that could be explored.

In any case, I feel like the concerns between the two extensibility options are different enough that they should be considered independently.

The main thing I see that they both share is the need (potentially) to show UI conditionally depending on the presence of other blocks.
